borsh-derive/Cargo.toml

```diff
@@ -18,6 +18,7 @@ proc-macro = true
 [dependencies]
 borsh-derive-internal = { path = "../borsh-derive-internal", version = "0.11.0" }
 borsh-schema-derive-internal = { path = "../borsh-schema-derive-internal", version = "0.11.0" }
-syn = {version = "1", features = ["full", "fold"] }
-proc-macro-crate = "0.1.5"
+syn = { version = "2.0.18", features = ["full", "fold"] }
```

Comment:
Unless you explicitly need at least a 2.0.18 version, use `2` instead to let other dependencies prefer some specific version and thus avoid duplication of the syn crate in the dependencies tree.